karyon3-core/src/main/java/com/netflix/karyon/Karyon.java

```diff
@@ -283,83 +320,90 @@ public LifecycleInjector start() {
         
-        if (isFeatureEnabled(KaryonFeatures.USE_DEFAULT_KARYON_MODULE)) 
-            apply(new DefaultKaryonModule());
+        final Logger LOG = LoggerFactory.getLogger(Karyon.class);
         
-        for (ModuleListTransformer transformer : transformers) {
-            modules = transformer.transform(Collections.unmodifiableList(modules));
-        }
+        // Create the main LifecycleManager to be used by all levels
+        final LifecycleManager manager = new LifecycleManager();
         
-        return LifecycleInjectorCreator.createInjector(new KaryonConfiguration() {
-            @Override
-            public List<Module> getModules() {
-                return modules;
-            }
-
-            @Override
-            public List<Module> getOverrideModules() {
-                return overrideModules;
-            }
-
-            @Override
-            public List<ModuleListProvider> getAutoModuleListProviders() {
-                return moduleProviders;
+        final KaryonFeatureSetImpl featureSet = new KaryonFeatureSetImpl(new IdentityHashMap<>(features));
+        
+        // Construct the injector using our override structure
+        try {
+            final Module coreModule = Modules.override(
+                ImmutableList.<Module>builder()
+                    .addAll(modules)
+                    .add(new LifecycleModule())
+                    .add(new ConditionalSupportModule())
+                    .add(new AbstractModule() {
+                        @Override
+                        protected void configure() {
+                            bind(KaryonFeatureSet.class).toInstance(featureSet);
+                            bind(LifecycleManager.class).toInstance(manager);
+                            
+                            Multibinder<String> profiles = Multibinder.newSetBinder(binder(), String.class, Profiles.class);
+                            for (String profile : Karyon.this.profiles) {
+                                profiles.addBinding().toInstance(profile);
+                            }
+                        }
+                    })
+                    .build())
+                .with(overrideModules);
+            
+            for (Element binding : Elements.getElements(coreModule)) {
+                LOG.debug("Binding : {}", binding);
             }
-
-            @Override
-            public Set<String> getProfiles() {
-                return profiles;
+            
+            Injector injector = Guice.createInjector(
+                stage,
+                coreModule,
+                new AbstractModule() {
+                    @Override
+                    protected void configure() {
+                        Set<Key<?>> boundKeys = ElementsEx.getAllBoundKeys(Elements.getElements(coreModule));
+                        Set<Key<?>> injectionKeys = ElementsEx.getAllInjectionKeys(Elements.getElements(coreModule));
+                        injectionKeys.removeAll(boundKeys);
```

Comment:
The presence of conditionals makes it so the catch-all factory isn't used.  These two features were really meant to solve completely different problems so I'm not sure linking them makes sense.  Conditionals are meant for context based decisions where the specific implementing classes are known whereas the AutoBinder is meant to be used with dynamic bindings such as creating proxies for archaius annotated interfaces.